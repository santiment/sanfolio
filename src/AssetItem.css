.assets-item {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 8px 15px;
  background: white;
  border-bottom: 1px solid #eee;
}

.assets-item .assets-item-title {
  display: flex;
  flex-direction: row;
  margin-right: 10px;
  width: 20%;
  min-width: 20%;
}

.assets-item .assets-item-avatar {
  width: 13px;
  min-width: 13px;
  height: 13px;
  border-radius: 50%;
  margin-right: 8px;
}

.assets-item .assets-item-details {
  flex: 1;
  display: flex;
  flex-direction: column;
  text-align: left;
  background-color: white;
  transition: 0.3s ease;
}

.assets-item .assets-item-graph {
  height: 45px;
}

.assets-item .assets-item-details .assets-item-details-price {
  color: green;
}

.assets-item .assets-item-details .assets-item-details-cap {
  color: #8a8a8a;
  font-weight: 300;
}

.asset-item-details-price-flash {
  background-color: yellow;
}
