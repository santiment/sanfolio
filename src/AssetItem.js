import React from 'react'
import './AssetItem.css'
import { formatNumber } from './utils/formatting'

const AssetAvatar = ({symbol}) => {
  return (
    <div
      className='assets-item-avatar'
      style={{backgroundColor: '#FF7300'}}
     />
  )
}

// FIXME:
const PRICE = 6500

const AssetItem = ({asset}) => {
  return (
    <div className='assets-item'>
      <div className='assets-item-title'>
        <AssetAvatar symbol={asset.title} />
        <strong>{asset.title}</strong>
      </div>
      <div className='assets-item-details'>
        <div className='assets-item-details-price'>{formatNumber(PRICE, 'USD')}</div>
        <div className='assets-item-details-cap'>{formatNumber(asset.cap, 'USD')}</div>
      </div>
      <div className='assets-item-graph'>
        graph;)
      </div>
    </div>
  )
}

export default AssetItem
