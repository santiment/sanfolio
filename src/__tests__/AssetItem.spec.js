/* eslint-env jasmine */
import AssetItem from '../AssetItem.js'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('AssetItem', () => {
  let wrapper
  const asset = {
    title: 'BTC',
    index: 0,
    cap: 1202021
  }

  beforeEach(() => {
    wrapper = shallow(<AssetItem asset={asset} />)
  })

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('render the DUMB component', () => {
    expect(wrapper.length).toEqual(1)
  })

  it('contains output', () => {
    expect(wrapper.find('strong').text()).toEqual(asset.title)
  })
})
