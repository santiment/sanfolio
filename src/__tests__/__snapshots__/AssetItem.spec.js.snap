// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetItem should render correctly 1`] = `
<div
  className="assets-item"
>
  <div
    className="assets-item-title"
  >
    <AssetAvatar
      symbol="BTC"
    />
    <strong>
      BTC
    </strong>
  </div>
  <div
    className="assets-item-details"
  >
    <div
      className="assets-item-details-price"
    >
      $6,500.00
    </div>
    <div
      className="assets-item-details-cap"
    >
      $1,202,021.00
    </div>
  </div>
  <div
    className="assets-item-graph"
  >
    graph;)
  </div>
</div>
`;
